### DISCUSSION TITLE
Typescript defines non-nullable relationship as nullable

### DISCUSSION THREAD

**Discussion by rickyhopkins** [prisma/prisma#9305] [Q&A]
I am building up a schema in prisma which is served up through an apollo server. I am really liking the nice integration with prisma, gql codegen, and apollo.

I have run into one issue that I believe is caused by incorrect typings coming from prisma on a non-nullable relationship.

The core of the problem relates to a piece of my schema that looks something like this:

```prisma
model Parent {
  id           String       @id @default(uuid())
  children Child[]
}

model Child {
  id           String       @id @default(uuid())
  parentId String
  parent   Parent @relation(fields: [parentId], references: [id])
}
```

My graphql schema now looks like this:

```graphql
type Parent {
	id: ID!
	children: [Child!]!
}

type Child {
	id: ID!
	parentId: ID!
	parent: Parent!
}
```

At this stage everyone is happy, until I try and implement my `Child.parent` resolver using [N+1 optimisation](https://www.prisma.io/docs/guides/performance-and-optimization/query-optimization-performance#solving-the-n1-problem)

```typescript
export const resolver = {
	Child: {
		parent: async ({id}, _, { prisma }) => {
			return prisma.child.findUnique({ where: { id: id } }).parent();
		},
	},
};
```

I get `Type 'Parent | null' is not assignable to type 'Parent'` this tells me that `prisma.child.findUnique({ where: { id: id } }).parent()` could be null. This should not be the case and it means that I either have to:

1. Handle the null scenario by throwing an error (I would have to do this for every non-nullable relationship.
2. Cast my `.parent()` call with `as Parent` (This is just bad)
3. Change my graphql schema to make the relationship optional.

Am I doing something wrong or missing something?

**Comment by ryands17** [comment]
This is a know issue with the fluent API that you can check here. For now, either add make it nullable or use `select` or `include` instead of the Fluent API.
https://github.com/prisma/prisma/issues/5715
https://github.com/prisma/prisma/issues/8929

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
